\documentclass[final]{beamer}
\usepackage{grffile}
\mode<presentation>{\usetheme{CambridgeUSPOL}}

\usepackage[utf8]{inputenc}
\usepackage{amsfonts}
\usepackage{amsmath}
\usepackage{natbib}
\usepackage{graphicx}
\usepackage{array,booktabs,tabularx}
\usepackage{epstopdf}
\usepackage{colortbl, xcolor}
% consider commenting me out ----------------------------------------------------
\usepackage{lipsum}
% -------------------------------------------------------------------------------
\newcolumntype{Z}{>{\centering\arraybackslash}X}

% rysunki
\usepackage{tikz}
\usepackage{ifthen}
\usepackage{xxcolor}
\usetikzlibrary{arrows}
\usetikzlibrary[topaths]
\usetikzlibrary{decorations.pathreplacing}
%\usepackage{times}\usefonttheme{professionalfonts}  % times is obsolete
\usefonttheme[onlymath]{serif}
\boldmath
\usepackage[orientation=portrait,size=a0,scale=1.4,debug]{beamerposter}                       % e.g. for DIN-A0 poster
%\usepackage[orientation=portrait,size=a1,scale=1.4,grid,debug]{beamerposter}                  % e.g. for DIN-A1 poster, with optional grid and debug output
%\usepackage[size=custom,width=200,height=120,scale=2,debug]{beamerposter}                     % e.g. for custom size poster
%\usepackage[orientation=portrait,size=a0,scale=1.0,printer=rwth-glossy-uv.df]{beamerposter}   % e.g. for DIN-A0 poster with rwth-glossy-uv printer check
% ...
%

\usecolortheme{seagull}
\useinnertheme{rectangles}
\setbeamercolor{item projected}{bg=darkred}
% \setbeamertemplate{enumerate items}[default]
\setbeamertemplate{caption}{\insertcaption} 
\setbeamertemplate{navigation symbols}{}
\setbeamercovered{transparent}
\setbeamercolor{block title}{fg=darkred}
\setbeamercolor{local structure}{fg=darkred}

\setbeamercolor*{enumerate item}{fg=darkred}
\setbeamercolor*{enumerate subitem}{fg=darkred}
\setbeamercolor*{enumerate subsubitem}{fg=darkred}

\setbeamercolor*{itemize item}{fg=darkred}
\setbeamercolor*{itemize subitem}{fg=darkred}
\setbeamercolor*{itemize subsubitem}{fg=darkred}

\newlength{\columnheight}
\setlength{\columnheight}{96cm}
\renewcommand{\thetable}{}
\def\andname{,}
\authornote{}

\renewcommand{\APACrefatitle}[2]{}
\renewcommand{\bibliographytypesize}{\footnotesize} 
\renewcommand{\APACrefYearMonthDay}[3]{%
  {\BBOP}{#1}
  {\BBCP}
}

\begin{document}

<<knitrIntro, echo = FALSE, message=FALSE,warning=FALSE>>=
source("my_ggplot_theme.R")

library(dplyr)
library(reshape2)
library(xtable)
library(biogram)

size_mod <- 4
cool_theme <- theme(plot.background=element_rect(fill = "transparent",
                                                 colour = "transparent"),
                    panel.grid.major = element_line(colour="lightgrey", linetype = "dashed"),
                    panel.background = element_rect(fill = "white",colour = "black"),
                    legend.background = element_rect(fill="NA"),
                    legend.position = "bottom",
                    axis.text = element_text(size=12 + size_mod),
                    axis.title.x = element_text(size=16 + size_mod, vjust = -1), 
                    axis.title.y = element_text(size=16 + size_mod, vjust = 1),
                    strip.text = element_text(size=17 + size_mod, face = "bold"),
                    strip.background = element_rect(fill="lightsteelblue", colour = "black"),
                    legend.text = element_text(size=13 + size_mod), 
                    legend.title = element_text(size=17 + size_mod),
                    plot.title = element_text(size=20 + size_mod))

@


\date{}
\author{Micha\l{} Burdukiewicz\inst{1}, John Doe\inst{2} and Jane Doe\inst{3}\\
\small{*michalburdukiewicz@gmail.com}}


\institute{\small{\textsuperscript{1}University of Wroc\l{}aw, Department of Genomics 

\textsuperscript{2}Wroc\l{}aw University of Something, Faculty of Something

\textsuperscript{3}Wroc\l{}aw University of Something, Faculty of Something Else}
}
}
\title{\huge biogram: a toolkit for biological n-gram analysis}

\begin{frame}
\begin{columns}
\begin{column}{.485\textwidth}
\begin{beamercolorbox}[center,wd=\textwidth]{postercolumn}
\begin{minipage}[T]{.95\textwidth}
\parbox[t][\columnheight]{\textwidth}
{
\begin{block}{Introduction}
\lipsum[1]

    \end{block}


\begin{block}{Tables in R}

\small{
       \begin{columns}[c] % the "c" option specifies center vertical alignment
    \column{.5\textwidth} 
<<echo = FALSE,message=FALSE,results='asis'>>=
sample_seq <- matrix(sample(c("A", "C", "G", "T"), 18, replace = TRUE), nrow = 3)
colnames(sample_seq) <- paste0("P", 1L:ncol(sample_seq))
rownames(sample_seq) <- paste0("S", 1L:nrow(sample_seq))
rws <- seq(1, nrow(sample_seq) - 1, by = 2)
col <- rep("\\rowcolor{white}", length(rws))

print(xtable(data.frame(sample_seq), caption = "Sample sequences.  S - sequence, P - postion.", digits = 0),
      booktabs = TRUE, add.to.row = list(pos = as.list(rws), command = col))

# unis <- count_ngrams(sample_seq, 1, 1L:4, pos = TRUE)
# unis <- data.frame(as.matrix(unis))[, 1L:7]
# print(xtable(unis, caption = "A fraction of possible unigrams with position information.", digits = 0), include.rownames = FALSE)
@
      
      
     % column designated by a command

    \column{.5\textwidth}
    
<<echo = FALSE,message=FALSE,results='asis'>>=
    
     unis <- count_ngrams(sample_seq, 1, c("A", "C", "G", "T"))
unis <- data.frame(as.matrix(unis))
colnames(unis) <- c("A", "C", "G", "T")
rownames(unis) <- paste0("S", 1L:nrow(sample_seq))
print(xtable(unis, caption = "Unigram counts.", digits = 0), include.rownames = TRUE,
      booktabs = TRUE, add.to.row = list(pos = as.list(rws), command = col))
@

    \end{columns}

}    
    \end{block}


\begin{block}{Equation}
    

\begin{center}
\scalebox{0.85}{
$
n_{\text{max}} = L \times m^n
$
}
\end{center}

\end{block}


\begin{block}{R plot}

<<echo = FALSE,message=FALSE,fig.align='center',fig.width=14,fig.height=6>>=
library(reshape2)
n <- 1:6
n_max <- 4^n

dat <- melt(sapply(2:5*5, function(i) i * n_max))
dat[["Var2"]] <- factor(dat[["Var2"]])
levels(dat[["Var2"]]) <- 2:5*5
colnames(dat) <- c("n", "L", "n_max")
ggplot(dat, aes(x = n, y = n_max, fill = L, col = L)) + 
  geom_line(linetype="dotted", size=1.5) + 
  geom_point(size=7, shape=21) +
  scale_x_continuous(breaks = 0L:6) +
  scale_y_continuous(expression(n["max"])) +
  my_theme

@

    \end{block}
    \vfill
    
    \begin{block}{Reducing number of n-grams}
    \begin{figure}
    \centering
    \includegraphics[width=0.5\columnwidth]{static_figure/example.png} 
    \end{figure}

A) Input data: peptides with a known status (e.g. amyloid/nonamyloid). 

B) Creation of an encoding using a combination of physicochemical properties (PP). 

C) Reduction of the amino acid alphabet according to an encoding. The number of possible n-grams is reduced, because $m$ is smaller (e.g. in this case $m$ is reduced from 20 to 6).
      
    \end{block}
    \vfill

}
\end{minipage}
\end{beamercolorbox}
\end{column}


%new column ------------------------------------------------------    

\begin{column}{.51\textwidth}
\begin{beamercolorbox}[center,wd=\textwidth]{postercolumn}
\begin{minipage}[T]{.95\textwidth}  
\parbox[t][\columnheight]{\textwidth}
{

\begin{block}{signalHsmm -  prediction of signal peptides}
Signal peptides are n-terminal guiding sequences with three distinguishable regions: n-, h- and c-region. Using the n-gram approach we created \textit{signalHsmm}, a software for prediction of signal peptides.

\medskip

    \textit{signalHsmm} has two models representing respectively proteins with and without signal peptides. The probabilities of both fits and predicted cleavage site constitute the software output.    
    \begin{figure}
    \centering
    \resizebox{32.5cm}{!}{%
    \begin{tikzpicture}[->,>=stealth',shorten >=2pt,auto,node distance=9.5cm, thick]
      \tikzstyle{line} = [draw=black, color=blue!30!black!50, line width=4.5mm, -latex']
      \tikzstyle{main node} = [circle,fill=blue!20,draw, minimum size = 2.2cm, font=\itshape,
         align=center,  top color=white, bottom color=blue!50!black!70 ] %font=\sffamily\small\bfseries,
      %nodes
      \node[main node]          	(start') 	[]						{Start};	     
      \node[main node, bottom color=purple!70!black!70] 	(nregion') 	[right of=start',xshift=-5mm, yshift=15mm] 	{n-region};
      \node[main node, bottom color=pink!70!black!70] 	(hregion') 	[right of=nregion',xshift=-5mm,yshift=15mm] 	{h-region};
      \node[main node, bottom color=gray!70!black!70] 	(cregion') 	[right of=hregion',xshift=-5mm,yshift=-15mm] 	{c-region};
      \node[main node, bottom color=green!70!black!70] 	(mature') 	[right of=cregion',xshift=-5mm, yshift=-15mm] 	{Mature protein};
      
      %lines
      \path [line] (start')   edge node [left, color=black] {} (nregion');
      \path [line] (nregion') edge node [below, color=black] { } (hregion');
      \path [line] (hregion') edge node [below, color=black] { } (cregion');
      \path [line] (cregion') edge node [left, color=black] { } (mature');
      \draw [line] (start') to[out=340,in=200] (mature');
    \end{tikzpicture} }
    \end{figure}


    \end{block}
    \vfill   

\begin{block}{signalHsmm benchmark}
\begin{table}[ht]
\centering

\begin{tabular}{rllll}
  \toprule
 & Sensitivity & Specificity & MCC & AUC \\ 
  \midrule
signalP 4.1 (no tm) \citep{2011petersensignalp} & 0.8235 & \textbf{0.9100} & 0.6872 & 0.8667 \\ 
   \rowcolor{white}signalP 4.1 (tm) \citep{2011petersensignalp} & 0.6471 & 0.9431 & 0.6196 & 0.7951 \\
   signalHsmm & \textbf{0.9804} & 0.8720 & \textbf{0.7409} & \textbf{0.9262} \\ 
   \rowcolor{white}signalHsmm (raw aa) & 0.8431 & 0.9005 & 0.6853 & 0.8718 \\ 
   \bottomrule
\end{tabular}
\end{table}

Comparison of performance measures for different classifiers according to singal peptide-containing proteins from members of \textit{Plasmodiidae}. 

\medskip

Thanks to the usage of reduced amino acid alphabet, \textit{signalHsmm} better recognizes signal peptides belonging to \textit{Plasmodiidae} which are characterized by an atypical amino acid composition.


\end{block}
\vfill

\begin{block}{AmyloGram}
Amyloids are proteins associated with the number of clinical disorders (e.g., Alzheimer's, Creutzfeldt-Jakob?s and Huntington?s diseases). We created \textit{AmyloGram}, n-based predictor of amyloidogenicity using decision rules extracted by random forests.

\medskip

\begin{table}[ht]
\centering

\begin{tabular}{ccccc}
  \toprule
Classifier & AUC & MCC & Sensitivity & Specificity \\ 
  \midrule
AmyloGram & \textbf{0.8972} & \textbf{0.6307} & 0.8658 & 0.7889 \\ 
  \rowcolor{white}PASTA \citep{walsh_pasta_2014} & 0.8550 & 0.4291 & 0.3826 & 0.9519 \\ 
   FoldAmyloid \citep{garbuzynskiy_foldamyloid:_2010} & 0.7351 & 0.4526 & 0.7517 & 0.7185 \\ 
  \rowcolor{white}APPNN \citep{familia_prediction_2015} & 0.8343 & 0.5823 & \textbf{0.8859} & 0.7222 \\ 
   \bottomrule
\end{tabular}
\end{table}

\end{block}
\vfill



\begin{block}{Summary and funding}
The n-gram analysis creates versatile classifiers able to extract more universal decision rules (e.g. \textit{signalHsmm}, which is able to also predict signal peptides in atypical organisms) or better detect specific proteins. Nonetheless, despite computational quickness provided by the QuiPT method, curse of dimensionality limits n-gram methods to the analysis of shorter sequences.

\bigskip

Our software is avaible as web-servers:

\begin{enumerate}

\item \textit{signalHsmm} web-server: \url{smorfland.uni.wroc.pl/signalHsmm}.

\item \textit{AmyloGram} web-server: \url{smorfland.uni.wroc.pl/amylogram}.

\end{enumerate}

\bigskip

This research was partially funded by the KNOW Consortium and National Science Center (2015/17/N/NZ2/01845).

\end{block}
\vfill

 \begin{block}{Bibliography}
  \tiny{
  \bibliographystyle{apalike}
  \bibliography{references}
  }
  \end{block}
  \vfill  


}
\end{minipage}
\end{beamercolorbox}
\end{column}
\end{columns}  
\end{frame}
\end{document}